logging/logging.go

```diff
@@ -0,0 +1,77 @@
+// Package logging package contains functionality for viam-server logging.
+package logging
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Level is an enum of log levels. Its value can be `DEBUG`, `INFO`, `WARN` or `ERROR`.
+type Level int
+
+const (
+	// This numbering scheme serves two purposes:
+	//   - A statement is logged if its log level matches or exceeds the configured level. I.e:
+	//     Statement(WARN) >= LogConfig(INFO) would be logged because "1" > "0".
+	//   - INFO is the default level. So we start counting at DEBUG=-1 such that INFO is given Go's
+	//     zero-value.
+
+	// DEBUG log level.
+	DEBUG Level = iota - 1
+	// INFO log level.
+	INFO
+	// WARN log level.
+	WARN
+	// ERROR log level.
+	ERROR
+)
```

Comment:
[q] no plans to add the go-specific and side-effectful PANIC and FATAL right?

Reply:
Good callout. I didn't have plans to add panic/fatal, but we do have code (examples + non-viam server CLI programs) that do use those. I did say I was going to keep the same logging APIs for the rdk, so I think the plan will be be:
* Continue having a `logger.Panic`/`Panicf`/`Fatal`/`Fatalf` (maybe others) library methods.
* Keep the configuration the way it is with DEBUG/INFO/WARN/ERROR.
* If we find out one day that logging all errors in some cases is too chatty, we can toss in fatal as another configuration level.